src/tests/validation.test.ts

```diff
@@ -371,4 +374,88 @@ suite('ASL context-aware validation', () => {
         })
       })
     })
+
+    suite('Additional properties that are not valid', async () => {
+        test('Shows diagnostics for additional invalid properties of a given state', async () => {
+            await testValidations({
+                json: documentInvalidPropertiesState,
+                diagnostics: [
+                    {
+                        message: MESSAGES.INVALID_PROPERTY_NAME,
+                        start: [7, 10],
+                        end: [7, 29]
+                    },
+                    {
+                        message: MESSAGES.INVALID_PROPERTY_NAME,
+                        start: [8, 10],
+                        end: [8, 29]
+                    }
+                ],
+                filterMessages: [MESSAGES.UNREACHABLE_STATE]
+            })
+        })
+
+        test('Shows diagnostics for additional invalid properties within Catch block', async () => {
+            await testValidations({
+                json: documentInvalidPropertiesCatch,
+                diagnostics: [
+                    {
+                        message: MESSAGES.INVALID_PROPERTY_NAME,
+                        start: [12, 20],
+                        end: [12, 32]
+                    },
+                    {
+                        message: MESSAGES.INVALID_PROPERTY_NAME,
+                        start: [19, 20],
+                        end: [19, 32]
+                    },
+                    {
+                        message: MESSAGES.INVALID_PROPERTY_NAME,
+                        start: [20, 20],
+                        end: [20, 34]
+                    }
+                ],
+                filterMessages: [MESSAGES.UNREACHABLE_STATE]
+            })
+        })
+
+        test('Shows diagnostics for additional invalid properties within Choice state', async () => {
+            await testValidations({
+                json: documentInvalidPropertiesChoices,
+                diagnostics: [
+                    {
+                        message: MESSAGES.INVALID_PROPERTY_NAME,
+                        start: [17, 22],
+                        end: [17, 40]
+                    },
+                    {
+                        message: MESSAGES.MUTUALLY_EXCLUSIVE_PROPERTIES + 'StringEquals, NumericGreaterThanEquals',
+                        start: [15, 22],
```

Comment:
Does the order of the diagnostics matter for the test to pass? If so is there a reason we don't return them in strictly ascending order?

Reply:
The order of diagnostics doesn't matter. The `testValidations` function does check the diagnostics in the order specified (It shouldn't). I'd prefer to update it in a separate ticket since it is a preexisting issue that doesn't influence the functionality.

Reply:
Sure, are you able to create the ticket and then resolve this comment?

Reply:
https://github.com/aws/amazon-states-language-service/issues/22